x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/ClientVersion.java

```diff
@@ -71,23 +72,36 @@
         CURRENT = extractVersion(url);
     }
 
+    // There are three main types of provided URLs:
+    // (1) a file URL: file:<path><FS separator><driver name>.jar
+    // (2) jar file URL pointing to a JAR file: jar:<sub-url><separator><driver name>.jar!/
+    // (3) jar file URL pointing to a JAR file entry (likely a fat JAR, but other types are possible): jar:<sub-url>!/driver name>.jar!/
     static SqlVersion extractVersion(URL url) {
+        Manifest manifest = null;
         String urlStr = url.toString();
         if (urlStr.endsWith(".jar") || urlStr.endsWith(".jar!/")) {
             try {
                 URLConnection conn = url.openConnection();
-                conn.setUseCaches(false);
```

Comment:
I think there was a reason for that, but can't remember. @astefan do you ?

Reply:
I didn't notice this change when I reviewed this. Good point.
I _think_ I added that because the temporary jars that were created during the test couldn't be cleaned-up properly. And I would keep the `setUseCaches(false)` as is. I was getting something like this:

```
java.io.IOException: Could not remove the following files (in the order of attempts):
   C:\Users\AppData\Local\Temp\org.elasticsearch.xpack.sql.client.VersionTests_123F282E30B0C58F-001\tempDir-002\foo.jar: java.nio.file.FileSystemException: C:\Users\AppData\Local\Temp\org.elasticsearch.xpack.sql.client.VersionTests_123F282E30B0C58F-001\tempDir-002\foo.jar: The process cannot access the file because it is being used by another process.
   C:\Users\AppData\Local\Temp\org.elasticsearch.xpack.sql.client.VersionTests_123F282E30B0C58F-001\tempDir-002: java.nio.file.DirectoryNotEmptyException: C:\Users\AppData\Local\Temp\org.elasticsearch.xpack.sql.client.VersionTests_123F282E30B0C58F-001\tempDir-002
   C:\Users\AppData\Local\Temp\org.elasticsearch.xpack.sql.client.VersionTests_123F282E30B0C58F-001: java.nio.file.DirectoryNotEmptyException: C:\Users\AppData\Local\Temp\org.elasticsearch.xpack.sql.client.VersionTests_123F282E30B0C58F-001
	at __randomizedtesting.SeedInfo.seed([123F282E30B0C58F]:0)
	at org.apache.lucene.util.IOUtils.rm(IOUtils.java:319)
	at org.apache.lucene.util.TestRuleTemporaryFilesCleanup.afterAlways(TestRuleTemporaryFilesCleanup.java:216)
	at com.carrotsearch.randomizedtesting.rules.TestRuleAdapter$1.afterAlways(TestRuleAdapter.java:31)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:43)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Reply:
Avoid file locking, at least on Windows.

Reply:
Not sure why my previous comment doesn't show up but it's because it avoids file locking on Windows as it frees the stream early on.

Reply:
Good catch, thanks @matriv. It was unintentional. I've put it back.